blog/views.py

```diff
@@ -0,0 +1,41 @@
+from django.contrib import messages
+from django.contrib.auth.decorators import login_required
+from django.core.paginator import Paginator
+from django.shortcuts import render, redirect
+from django.views import generic
+
+from blog.models import Post, Commentary
+
+
+def index(request):
+    posts = Post.objects.all().order_by("created_time")
+    paginator = Paginator(posts, 5)
+    page = request.GET.get("page")
+    page_obj = paginator.get_page(page)
+    context = {
+        "posts": page_obj,
+    }
+    return render(request,
+                  "blog/index.html",
+                  context=context, )
+
+
+class PostDetailView(generic.DetailView):
+    model = Post
```

Comment:
For the `PostDetailView`, consider adding a `queryset` or `get_queryset` method to filter only for published posts, unless you want to allow viewing of all posts.